### PR TITLE
Updated Oracle query to use rule based optimizer

### PR DESCRIPTION
**Pull request by aidenfoxx**
This drastically improves performance for data dictionary queries. See https://github.com/knex/knex-schema-inspector/pull/108

### CODE REVIEW

File: packages/schema/src/dialects/oracledb.ts [hunk 1]
```diff
@@ -45,37 +45,41 @@ export default class Oracle extends KnexOracle implements SchemaInspector {
...
 				"c"."DATA_DEFAULT" "default_value",
 				"c"."NULLABLE" "is_nullable",
 				"c"."DATA_TYPE" "data_type",
 				"c"."DATA_PRECISION" "numeric_precision",
 				"c"."DATA_SCALE" "numeric_scale",
-				"ct"."CONSTRAINT_TYPE" "column_key",
+				"uc"."CONSTRAINT_TYPE" "column_key",
 				"c"."CHAR_LENGTH" "max_length",
-				"c"."VIRTUAL_COLUMN" "is_generated"
+				"c"."VIRTUAL_COLUMN" "is_generated"
```
Comment (aidenfoxx) [inline packages/schema/src/dialects/oracledb.ts:62]:
```suggestion
				"c"."VIRTUAL_COLUMN" "is_generated"
```

File: packages/schema/src/dialects/oracledb.ts [hunk 2]
```diff
@@ -45,37 +45,41 @@ export default class Oracle extends KnexOracle implements SchemaInspector {
...
 				"c"."CHAR_LENGTH" "max_length",
 				"c"."VIRTUAL_COLUMN" "is_generated"
 			FROM "USER_TAB_COLS" "c"
-			LEFT JOIN "uc" "ct"
-				ON "c"."TABLE_NAME" = "ct"."TABLE_NAME"
-				AND "c"."COLUMN_NAME" = "ct"."COLUMN_NAME"
-				AND "ct"."CONSTRAINT_COUNT" = 1
-			WHERE "c"."HIDDEN_COLUMN" = 'NO'
+			LEFT JOIN (
+				SELECT /*+ rule */
```
Comment (aidenfoxx) [inline packages/schema/src/dialects/oracledb.ts:65]:
```suggestion
				SELECT /*+ RULE */
```

**Comment by rijkvanzanten** [comment]
This ready for review/merge @aidenfoxx? 🙂

**Comment by aidenfoxx** [comment]
@rijkvanzanten Well, since rule based optimization is technically depreciated (though I doubt it will ever actually be removed), it would probably be better to achieve the same result through the right optimization hints. Gonna see if I can figure that out first otherwise we can probably live with the rule based optimizer.

**Comment by rijkvanzanten** [comment]
Copy that!

**Pull request closed without merging**